### ISSUE TITLE
Vite compatibility

### ISSUE THREAD

**Issue by ksewo** [Camberi/firecms#147]
Hello!
It looks like firecms isn't compatible with [vite](https://vitejs.dev/), because of firecms usage of react-csv, which has Link.js file instead of .jsx extension and vite is using js loader instead of jsx loader for it (for performance purposes).

Related react-csv issue:
https://github.com/react-csv/react-csv/issues/268

It looks like this library is unmaintained and therefore should be replaced.

**Comment by ksewo** [comment]
@fgatti675 Thanks for reply.

It seems like firebaseui could be removed, it's only used for these login boxes, which shouldn't be really that hard to implement. I think that removing it and rewriting will be better idea. Social media login buttons should be easy to implement, with email/password it will be harder, but making it only sign-in (accounts can be created from firestore console) will make it lot easier.

Please let me know what you think about the idea of removing firebaseui, it looks like it's causing more harm than good.
What features would be needed in new auth page?

For react-csv there are two ways of fixing it:
1st one is forking the react-csv library and changing the extension to Link.js to .jsx and making firecms use forked library instead of npm one.
2nd is using other library capable of handling csv files

Which one would be better? I'll probably try submit PR in my free time.

If somebody have problem with react-csv library because of it using .js extension for jsx file, create your own fork and download it in your project (or use mine: https://github.com/ksewo/react-csv).

And updating your package.json to use this package in firecms instead:

For yarn:
Adding it in package.json:
```json
  "resolutions": {
    "react-csv": "github:ksewo/react-csv"
  }
```

For pnpm
Adding it in package.json:
```json
"pnpm": {
  "overrides": {
    "react-csv": "github:ksewo/react-csv"
  }
}
```
I recommend you using your own fork with this fix.

**Comment by fgatti675** [comment]
Hi @ksewo 
I would rather not fork a package and need to maintain it afterwards.
There are great csv packages in npm we can use, such as 
https://www.npmjs.com/package/csv
Wanna give it a try? :)

**Comment by NazimHAli** [comment]
Wish I read this thread an hour ago, have been trying to get it to work with Vite. I agree that `firebaseui` should be removed, it's just bloat and ended up removing them in another project. I just implemented the login forms with Svelte yesterday.

**Comment by fgatti675** [comment]
`firebaseui` is quite a pain to be honest.
But it provides a lot of functionality with very little effort.
In the best case we would create an independent library based on Firebase Auth, React, MUI 5 and all the login options
(maybe even without MUI 5).
I would start implementing it as part of this project and think about detaching it later.
What do you guys think?

**Comment by ksewo** [comment]
@fgatti675 That seems great. It's really just bloat and it's causing many problems, simple buttons for login with social medias + login form for username/password login should be enough.

@NazimHAli 
If you're struggling with vite, you can use craco with craco-esbuild (vite is using esbuild under the hood) so the build time should be similiar.

**Comment by NazimHAli** [comment]
> @fgatti675 That seems great. It's really just bloat and it's causing many problems, simple buttons for login with social medias + login form for username/password login should be enough.
> 
> @NazimHAli If you're struggling with vite, you can use craco with craco-esbuild (vite is using esbuild under the hood) so the build time should be similiar.

Thank you, but I was trying to integrate it into a current application without turning it into a mono-repo. For now, I'll just keep it as a separate admin package.

**Comment by NazimHAli** [comment]
@fgatti675 I think if we had a minimal version, it would give everyone an easy way to customize/build it the way they need and easier to integrate with current applications. The current version could be a sort of provided optional/default theme.

**Comment by fgatti675** [comment]
Hi @ksewo I have good news for you
I have replaced the CSV export function and FirebaseUI with custom implementations.
It is not published in npm yet since it needs more testing, but it is pushed to the master branch in this repository.
Could you check the code and see if this version works with your projects?
You will need to clone the repository, run `yarn build` and then link your project using `yarn link` to the built library.
https://classic.yarnpkg.com/en/docs/cli/link/
Thanks!

**Comment by fgatti675** [comment]
Both updates are now released in version 1.0.0-beta6
